sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcOptions.scala

```diff
@@ -20,30 +20,33 @@ package org.apache.spark.sql.hive.orc
 import java.util.Locale
 
 import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
+import org.apache.spark.sql.internal.SQLConf
 
 /**
  * Options for the ORC data source.
  */
-private[orc] class OrcOptions(@transient private val parameters: CaseInsensitiveMap[String])
+private[orc] class OrcOptions(
+    @transient private val parameters: CaseInsensitiveMap[String],
+    @transient private val sqlConf: SQLConf)
   extends Serializable {
 
   import OrcOptions._
 
-  def this(parameters: Map[String, String]) = this(CaseInsensitiveMap(parameters))
+  def this(parameters: Map[String, String], sqlConf: SQLConf) =
+    this(CaseInsensitiveMap(parameters), sqlConf)
 
   /**
-   * Compression codec to use. By default snappy compression.
+   * Compression codec to use.
    * Acceptable values are defined in [[shortOrcCompressionCodecNames]].
    */
   val compressionCodec: String = {
-    // `orc.compress` is a ORC configuration. So, here we respect this as an option but
-    // `compression` has higher precedence than `orc.compress`. It means if both are set,
-    // we will use `compression`.
+    // `compression`, `orc.compress`, and `spark.sql.orc.compression.codec` is used in order.
     val orcCompressionConf = parameters.get(OrcRelation.ORC_COMPRESSION)
     val codecName = parameters
       .get("compression")
       .orElse(orcCompressionConf)
-      .getOrElse("snappy").toLowerCase(Locale.ROOT)
+      .getOrElse(sqlConf.orcCompressionCodec)
```

Comment:
Could we update the default values for consistency with Parquet one? :

https://github.com/apache/spark/blob/3c0c2d09ca89c6b6247137823169db17847dfae3/sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala#L520

https://github.com/apache/spark/blob/51620e288b5e0a7fffc3899c9deadabace28e6d7/python/pyspark/sql/readwriter.py#L855

Reply:
The default value is [snappy](https://github.com/apache/spark/pull/19055/files#diff-9a6b543db706f1a90f790783d6930a13R331), isn't it?
```
.createWithDefault("snappy")
```

Reply:
I was thinking like:

https://github.com/apache/spark/blob/51620e288b5e0a7fffc3899c9deadabace28e6d7/python/pyspark/sql/readwriter.py#L751-L753

Wouldn't we use the value set in `spark.sql.parquet.compression.codec` by default if `compression` is unset via option API?

Reply:
Actually, I thought the purpose of this configuration is rather for setting the default compression codec for ORC datasource ..

Reply:
Yes. This is the priority. If `compression` and `orc.compression` is unset via option, we use SQLConf.
`compression` -> `orc.compression` -> `spark.sql.orc.compression.codec`

Reply:
The main purpose of this PR is to support users to control ORC compression by using SQLConf, too.

Reply:
The default codec is unchanged and the priority is the same. Also, all previous user given options are respected.

Reply:
Ah, it looks I had to be clear. I meant fixing the comment for default value from `snappy` to `spark.sql.orc.compression.codec`.

Reply:
Thank you! I'll fix that.